### PR TITLE
feat: make attribute maxLength configurable

### PR DESCRIPTION
**Pull request by fdlk**
- [x] Migration

#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] Migration step added in case of breaking change
- [NA] User documentation updated (see next PR)
- [x] (If you have changed REST API interface) view-swagger.ftl updated
- [NA] Test plan template updated
- [x] Clean commits
- [x] Added Feature/Fix to release notes

### CODE REVIEW

**Review by tommydeboer — CHANGES_REQUESTED** [review]

File: molgenis-data/src/main/java/org/molgenis/data/meta/model/Attribute.java [hunk 1]
```diff
@@ -765,4 +767,13 @@ public Attribute getInversedBy() {
   public boolean isInversedBy() {
     return getInversedBy() != null;
   }
+
+  public Integer getMaxLength() {
```
[the remark this answers is not in the document]
Comment (fdlk) [reply molgenis-data/src/main/java/org/molgenis/data/meta/model/Attribute.java:771]: Unsure. AttributeType.getMaxLength() does not have the annotation, so all of the reading code is full of assumptions.

[the remark this answers is not in the document]
Comment (fdlk) [reply molgenis-data/src/main/java/org/molgenis/data/meta/model/Attribute.java:771]: I've added the annotation, let's see how angry sonar gets

**Comment by tommydeboer** [comment]
This might be a forms issue but there's an `undefined` in the error message: 

> Invalid string value 'Dirk de haan' for attribute 'lastName' of entity 'it_emx_autoid_testAutoId'.Value must be less than or equal to 5 characters (undefined)

**Comment by tommydeboer** [comment]
Man, this feature can be pretty dangerous if you don't know what you're doing! All data already present in the column will be silently truncated if you reduce the `maxLength`.

**Comment by tommydeboer** [comment]
The forms input for strings can't take strings longer than 255.

**Comment by tommydeboer** [comment]
Should we add `maxLength` to:
- EMX?
- Metadata Manager?

**Comment by tommydeboer** [comment]
Hmm maybe the Data Explorer should also truncate very long strings like it does with text fields.

**Comment by tommydeboer** [comment]
**Issue:** `maxLength` doesn't show up in the metadata manager
1. Import `it_emx_autoid`
2. Set `maxLength` of `lastName` to 10
3. Hit `/api/metadata/it_emx_autoid_testAutoId`

```json
"data": {
  "id": "aaaac6fqlz6kwivhuemmfuaaay",
  "label": "lastName",
  "name": "lastName",
  "sequenceNr": 4,
  "type": "string",
  "idAttribute": false,
  "labelAttribute": false,
  "nullable": true,
  "auto": false,
  "visible": true,
  "unique": false,
  "readOnly": false,
  "aggregatable": false
}
```

**Review by tommydeboer — APPROVED** [review]

**Pull request merged**